File: Class_member/Class_member02.py
class Cs:
    @staticmethod  # 위에 장식자 넣어주기 (규칙이므로 꼭 저이름대로 넣어줘야 함)
    def static_method():
        print("Static method")

    @classmethod  # 위에 장식자 넣어주기 (규칙이므로 꼭 저이름대로 넣어줘야 함)
    def class_method(cls):  # self 처럼 클래스 메소드에서는 cls로 첫번째 인자를 줌
        print("Class method")

    def instance_method(self):
        print("instance_method")


i = Cs()
Cs.static_method()  # 클래스 소속이므로 Cs.
Cs.class_method()  # 클래스 소속이므로 Cs.
i.instance_method()  # 인스턴스 소속이므로 i.
# python에서 클래스 멤버에 두가지 종류로 static, class 두가지 임


# Ruby는 클래스 멤버를 구분하기 위해서 '클래스.' 을 붙여주고
# Python 에서는 @staticmethod 나 @classmethod 를 붙여주고 추가적으로 class_method에는 첫번째 인자로 'cls'를 넣어준다
# 둘다 클래스 멤버, 인스턴스 멤버를 사용하는 경우에는 각각에 맞게 똑같이 써준다.

File: Class_member/Class_member02.rb
# 클래스 멤버 만들기
class Cs
    def Cs.class_method() # Cs.를 붙여서 Cs라는 클래스의 멤버라는 것을 분명히 해야함
        p 'Class method'
    end
    def instnace_method()
        p 'Instnace_method'
    end
end

i = Cs.new()
Cs.class_method() # class_method()는 Cs라는클래스 소속인 클래스 멤버라는 것임 
i.instnace_method() # instance_method는 i 라는 인스턴스의 소속인 인스턴스 멤버라는 것임
# Cs.instnace_method() # error
# i.class_method() # error
# 클래스와 클래스 메소드의 합으로
# 인스턴스와 인스턴스 메소드의 합으로 구성 해야 한다.